testdriver/src/testdriver/sendparliamentaryresults.bal

```diff
@@ -0,0 +1,114 @@
+import ballerina/io;
+import ballerina/http;
+import ballerina/time;
+import ballerina/runtime;
+import ballerina/encoding;
+
+const R_V = "R_V";
+const R_S = "R_S";
+const R_VS = "R_VS";
+const R_VSN = "R_VSN";
+const R_SC = "R_SC";
+const R_NC = "R_NC";
+const R_SCNC = "R_SCNC";
+
+const FINAL = "FINAL";
+
+function sendParliamentaryResults(string electionCode, http:Client rc, map<json>[] results) returns error? {
+
+    foreach map<json> result in results {
+
+        string resultType = result.'type.toString();
+        match resultType {
+            R_V => {
+                check updateByParty(result);
+                check updateSummary(result);
+                check feedResult(rc, electionCode, resultType, result.pd_code.toString(), result);
+            }
+            R_S => {
+                check updateByParty(result);
+                check feedResult(rc, electionCode, resultType, result.ed_code.toString(), result);
+            }
+            R_VS => {
+                check updateByParty(result);
+                check updateSummary(result);
+                check feedResult(rc, electionCode, resultType, FINAL, result);
+            }
+            R_VSN => {
+                check updateByParty(result);
+                check updateSummary(result);
+                check feedResult(rc, electionCode, resultType, FINAL, result);
+            }
+            R_SC => {
+                check feedResult(rc, electionCode, resultType, result.ed_code.toString(), result);
+            }
+            R_NC => {
+                check feedResult(rc, electionCode, resultType, FINAL, result);
+            }
+            R_SCNC => {
+                check feedResult(rc, electionCode, resultType, FINAL, result);
+            }
+        }
+        // delay a bit
+        runtime:sleep(sleeptime);
+    }
+}
+
+function updateByParty(map<json> result) returns error? {
+    // add missing info in test data: party_name and candidate for each party result if needed only (2019 set has it)
+    json[] pr = <json[]>result.by_party;
+    foreach int i in 0 ..< pr.length() {
+        map<json> onePr = <map<json>>pr[i];
+        if pr[i].party_name is string {
```

Comment:
We don't get to the vote percentage conversion if this evaluates to true, which seems to be the case always, since the data has `party_name` and it is a `string`.